nextcord/application_command.py

```diff
@@ -0,0 +1,883 @@
+
+    def _get_basic_application_payload(self) -> dict:
+        """Bare minimum payload that both Global and Guild commands can use."""
+        payload = super().payload
+        if self.type is not ApplicationCommandType.chat_input and "options" in payload:
+            payload.pop("options")
+        if self.default_permission is not None:
+            payload["default_permission"] = self.default_permission
+        return payload
+
+    @property
+    def payload(self) -> List[dict]:
+        """Returns a list of Discord "Application Command Structure" payloads.
+
+        Returns
+        -------
+        payloads: List[:class:`dict`]
+            Returns a list containing the global command payload, if enabled, and payloads corresponding to every guild
+            ID specified.
+        """
+        # TODO: This always returns a list, should it be "payloads"? Won't override subcommand payload though.
+        partial_payload = self._get_basic_application_payload()
+
+        ret = []
+        if self.is_guild:
+            for guild_id in self.guild_ids:
+                temp = partial_payload.copy()  # This shouldn't need to be a deep copy as guild_id is on the top layer.
+                temp["guild_id"] = guild_id
+                ret.append(temp)
+        if self.is_global:
+            ret.append(partial_payload)
+        return ret
+
+    def get_guild_payload(self, guild_id: int) -> Optional[dict]:
+        """Creates a payload specific to a guild.
+
+        Parameters
+        ----------
+        guild_id: :class:`int`
+            Guild ID to create the payload for. Must be added to the ApplicationCommand.
+
+        Returns
+        -------
+        Optional[:class:`dict`]
+            A Discord "Application Command Structure" payload specifically for a given Guild ID. Returns None if the
+            guild ID specified isn't added.
+        """
+        if not self.is_guild or guild_id not in self.guild_ids:
+            return None
+        partial_payload = self._get_basic_application_payload()
+        partial_payload["guild_id"] = guild_id
+        return partial_payload
+
+    @property
+    def global_payload(self) -> dict:
+        """Creates a Discord "Application Command Structure" payload for a Global command."""
+        if not self.is_global:
+            raise NotImplementedError  # TODO: Make a proper error.
+        return self._get_basic_application_payload()
+
+    @property
+    def is_guild(self) -> bool:
+        return self._is_guild
+
+    @property
+    def is_global(self) -> bool:
+        return self._is_global
+
+    def get_signature(self, guild_id: Optional[int]) -> Optional[Tuple[str, int, Optional[int]]]:
+        """Returns a basic signature for a given Guild ID. If None is given, then it is assumed Global."""
+        if (guild_id is None and self.is_global) or (guild_id in self.guild_ids):
+            return self.name, self.type.value, guild_id
+        else:
```

Comment:
Personal opinion, don't have to but I'd get rid of the else and unindent the return